--- hdfeos5/samples/he5_pt_writedata.c.orig	2007-03-29 18:58:21.000000000 +0200
+++ hdfeos5/samples/he5_pt_writedata.c	2012-12-23 18:36:44.438035334 +0100
@@ -16,7 +16,7 @@
 
   typedef struct 
 	{
-	  char      Label[8];
+	  char      Label[9];
 	  double    Lon;
 	  double    Lat;
 	  int       Date;
--- hdfeos5/testdrivers/point/TestPoint.c.orig	2007-03-29 18:58:28.000000000 +0200
+++ hdfeos5/testdrivers/point/TestPoint.c	2012-12-23 18:45:28.971357733 +0100
@@ -79,7 +79,7 @@
   
   typedef struct
 	{
-	  char        label[8];
+	  char        label[9];
 	  double      lon;
 	  double      lat;
 	  int         date;
