sizeof(char*) obviously doesn't work for C string...
--- hdf-eos5-2.0/src/GDapi.c.orig	2024-12-15 09:28:35.178273464 +0100
+++ hdf-eos5-2.0/src/GDapi.c	2024-12-15 09:37:32.792027627 +0100
@@ -20,6 +20,8 @@
 #include <stdio.h>
 #include <HE5_GctpFunc.h>
 
+#define HE5_EN_FILENAME_LEN 300
+
 #define	HE5_GRIDOFFSET 671088642    /* (2**26)*10 + 2  */
 #define HE5_NGRID            400
 #define HE5_NGRIDREGN        512
@@ -15025,8 +15027,8 @@ HE5_GDll2ij(int projcode, int zonecode,
   if (projcode != HE5_GCTP_GEO)
     {
 
-      eastFile = (char *) malloc(300*sizeof(char));
-      northFile = (char *) malloc(300*sizeof(char));
+      eastFile = (char *) malloc(HE5_EN_FILENAME_LEN*sizeof(char));
+      northFile = (char *) malloc(HE5_EN_FILENAME_LEN*sizeof(char));
 
       HE5_GDgetEastAndNorthFiles(eastFile,northFile);
 
@@ -29957,8 +29959,8 @@ void HE5_GDgetEastAndNorthFiles(char *ea
 	lastables[(int)strlen(ptr)] = '\0';
         /*strcpy(lastables, ptr);*/
 
-        snprintf( eastFile, sizeof(eastFile), "%s/nad27sp_be", lastables );    /* use big endian binary files */
-        snprintf( northFile,sizeof(northFile), "%s/nad83sp_be", lastables );
+        snprintf( eastFile, HE5_EN_FILENAME_LEN, "%s/nad27sp_be", lastables );    /* use big endian binary files */
+        snprintf( northFile,HE5_EN_FILENAME_LEN, "%s/nad83sp_be", lastables );
 
 
     }
