--- hdf-eos5-2.0/src/Makefile.am.orig	2024-12-15 09:11:05.253961392 +0100
+++ hdf-eos5-2.0/src/Makefile.am	2024-12-15 09:11:46.330405529 +0100
@@ -10,10 +10,10 @@ AM_CPPFLAGS=-I$(top_srcdir)/include/
 
 # Set LDFLAGS to allow the HDF-EOS library to use extern variables from
 # HDF5
-LDFLAGS=-Wl,-single_module
 
 # Build HDF-EOS5
 lib_LTLIBRARIES=libhe5_hdfeos.la
+libhe5_hdfeos_la_LIBADD=$(top_builddir)/gctp/src/libhe5_Gctp.la -lhdf5_hl -lhdf5
 
 # Source files for library
 libhe5_hdfeos_la_SOURCES=EHapi.c GDapi.c PTapi.c SWapi.c TSapi.c ZAapi.c HE5common.c
--- hdf-eos5-2.0/gctp/src/Makefile.am.orig	2021-04-22 01:06:56.000000000 +0200
+++ hdf-eos5-2.0/gctp/src/Makefile.am	2024-12-15 09:12:37.443461959 +0100
@@ -4,7 +4,7 @@
 include $(top_srcdir)/config/include.am
 
 # Library to build
-lib_LTLIBRARIES = libGctp.la
+lib_LTLIBRARIES = libhe5_Gctp.la
 
 ## Normally DEFAULT_INCLUDES is supplied by Automake, but one of the
 ## directories included by default is $(top_builddir)/include, which
@@ -17,7 +17,7 @@ DEFAULT_INCLUDES = -I. -I$(srcdir)
 AM_CPPFLAGS=-I$(srcdir)/../include/
 
 # Library source files
-libGctp_la_SOURCES = gctp.c alberfor.c alberinv.c alconfor.c alconinv.c      \
+libhe5_Gctp_la_SOURCES = gctp.c alberfor.c alberinv.c alconfor.c alconinv.c      \
           azimfor.c aziminv.c bceafor.c bceainv.c br_gctp.c ceafor.c         \
           ceainv.c cproj.c eqconfor.c eqconinv.c equifor.c equiinv.c         \
           for_init.c gnomfor.c gnominv.c goodfor.c goodinv.c gvnspfor.c      \
--- hdfeos5/config/include.am.orig	2007-03-29 18:59:12.000000000 +0200
+++ hdfeos5/config/include.am	2012-12-23 16:25:47.688199095 +0100
@@ -4,5 +4,5 @@
 ## shared definitions.
 
 LIBHDFEOS5=$(top_builddir)/src/libhe5_hdfeos.la
-LIBGCTP=$(top_builddir)/gctp/src/libGctp.la
+LIBGCTP=$(top_builddir)/gctp/src/libhe5_Gctp.la
 
